torch/utils/data/standard_pipes.ipynb

```diff
@@ -0,0 +1,1006 @@
+{
+ "metadata": {
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.6.10"
+  },
+  "orig_nbformat": 2,
+  "kernelspec": {
+   "name": "python3610jvsc74a57bd0eb5e09632d6ea1cbf3eb9da7e37b7cf581db5ed13074b21cc44e159dc62acdab",
+   "display_name": "Python 3.6.10 64-bit"
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 2,
+ "cells": [
+  {
+   "source": [
+    "## Standard flow control and data processing DataPipes"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from torch.utils.data import IterDataPipe"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Example IterDataPipe\n",
+    "class ExampleIterPipe(IterDataPipe):\n",
+    "    def __init__(self, range = 20):\n",
+    "        self.range = range\n",
+    "    def __iter__(self):\n",
+    "        for i in range(self.range):\n",
+    "            yield i"
+   ]
+  },
+  {
+   "source": [
+    "## Concat\n",
+    "\n",
+    "Function: `concat`\n",
+    "\n",
+    "Description: Returns DataPipes with elements from the first datapipe following by elements from second datapipes\n",
+    "\n",
+    "Alternatives:\n",
+    "\n",
+    "    `dp = dp + dp2`\n",
```

Comment:
I suggest we drop "+" actually and just support "concat" only. If we're operating on datafram-ish pipe, then I can write `dp["a"] + dp["b"]` which means adding column values. But then `dp + dp` would double the number of rows. It's confusing and better just avoid this overload, especially given that it's not that common imho (is it?)